src/realm/object-store/collection.hpp

```diff
@@ -35,11 +35,13 @@ namespace _impl {
 class ListNotifier;
 }
 
+
 namespace object_store {
 class Collection {
 public:
     Collection(PropertyType type) noexcept;
     Collection(const Object& parent_obj, const Property* prop);
+    Collection(Obj& parent_obj, ColKey col, std::shared_ptr<Realm> r);
```

Comment:
This is really hacky, I need to find a better way for storing` obj, realm` and `col_key ` for a collection. This constructor can be used only with mixed, since they are the only types that support dynamic collections inside them. In all the other cases, the schema must be explicitly defined.